# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1506369771 25200
#      Mon Sep 25 13:02:51 2017 -0700
# Node ID b47cbbfff5012518aaef1b7a13f10baf9e27d38c
# Parent  71dad5be765a5568d83073bcb916217abf265146
test: restore showing contents of fntests.log

* test/module.mk (run-octave-tests): Change directory to run the test suite
in a subshell so relative paths are correct for the rest of the sequence.

diff --git a/test/module.mk b/test/module.mk
--- a/test/module.mk
+++ b/test/module.mk
@@ -72,17 +72,17 @@ include %reldir%/nest/module.mk
 include %reldir%/publish/module.mk
 
 ALL_LOCAL_TARGETS += %reldir%/.gdbinit
 
 %reldir%/.gdbinit: etc/gdbinit
 	@$(gdbinit_install_rule)
 
 define run-octave-tests
-  cd test && $(SHELL) ../run-octave $(RUN_OCTAVE_OPTIONS) $(1) --norc --silent --no-history $(abs_top_srcdir)/%reldir%/fntests.m $(abs_top_srcdir)/test; \
+  ( cd %reldir% && $(SHELL) ../run-octave $(RUN_OCTAVE_OPTIONS) $(1) --norc --silent --no-history $(abs_top_srcdir)/%reldir%/fntests.m $(abs_top_srcdir)/%reldir% ); \
   if $(AM_V_P); then \
     echo ""; \
     if [ -f %reldir%/fntests.log ]; then \
       echo "Contents of %reldir%/fntests.log:"; \
       echo ""; \
       $(AWK) -f $(srcdir)/%reldir%/show-failures.awk %reldir%/fntests.log; \
     else \
       echo "%reldir%/fntests.log is missing!"; \
